code/game/objects/items/flag.dm

```diff
@@ -23,11 +31,16 @@
 
 /obj/item/flag/update_icon()
 	overlays.Cut()
+	icon_state = replacetext(icon_state, "_rolled", "") //This will totally not break in the future
```

Comment:
There's a better way of doing this, will play around with it tomorrow when I'm not as sleepy

Reply:
icon_state = initial(icon_state)?

Reply:
Unfortunately that breaks chameleon flags, current workaround is to have a second var that is set to `icon_state`, and use that for the icon

Reply:
Fair enough, I totally forgot about the chams
Hearts work the same way